tomviz/DataSource.cxx

```diff
@@ -471,35 +514,54 @@ void DataSource::operate(Operator* op)
 {
   Q_ASSERT(op);
 
-  vtkTrivialProducer* tp = vtkTrivialProducer::SafeDownCast(
-    this->Internals->Producer->GetClientSideObject());
-  Q_ASSERT(tp);
-  if (op->transform(tp->GetOutputDataObject(0))) {
-    this->dataModified();
+  // If we are currently executing the pipeline, just add the operator
+  if (this->Internals->Future != nullptr &&
+      this->Internals->Future->isRunning()) {
+    this->Internals->Future->addOperator(op);
+  }
+  // We need to initiate a new run
+  else {
+    vtkDataObject* copy = this->copyData();
```

Comment:
Yes, we need to copy it so we can fall back to the previous copy if the user cancels an operation ( which is coming ...)